### PR TITLE
[MacOS] Fix bugzilla58779

### PR DESCRIPTION
**Pull request by RogerHardiman**
### Description of Change ###

Xamarin Forms Mac OS X bug fix.
DisplayActionSheet shows a list of items the user can select from.
If the list is long (e.g. 15 items) the Popup went off the bottom of the Mac Desktop.
Not all items can be selected.
The cancel button goes off the bottom of the screen

### Bugs Fixed ###

Bugzilla 58779

### API Changes ###

None

### Behavioral Changes ###

A NSScrollView is used to allow long lists of items to fit on the Mac Desktop

### PR Checklist ###

- [X] Has tests (if omitted, state reason in description)
- [X] Rebased on top of master at time of PR
- [X] Changes adhere to coding standard
- [X] Consolidate commits as makes sense

### CODE REVIEW

**Comment by samhouts** [comment]
build

File: Xamarin.Forms.Platform.MacOS/Platform.cs [hunk 1]
```diff
@@ -43,7 +43,16 @@ internal Platform()
 				var alert = NSAlert.WithMessage(arguments.Title, arguments.Cancel, arguments.Destruction, null, "");
 				if (arguments.Buttons != null)
 				{
-					alert.AccessoryView = GetExtraButton(arguments);
+					NSView extraButtons = GetExtraButton(arguments);
+					if (extraButtons.Frame.Height > 400) {
```
Comment (rmarinho) [inline Xamarin.Forms.Platform.MacOS/Platform.cs]: we need something better instead of a magic 400 ..

Comment (RogerHardiman) [reply Xamarin.Forms.Platform.MacOS/Platform.cs]: I agree that we need something better than a magic number.
So I'll go with 60% of the mac desktop screen height.
Is there something in Platform.cs that already knows the size of the mac desktop?

Comment (RogerHardiman) [reply Xamarin.Forms.Platform.MacOS/Platform.cs]: Is this OK?
'''
int maxScrollHeight = (int)(0.6 * NSScreen.MainScreen.Frame.Height);
'''

**Comment by RogerHardiman** [comment]
Changes made

**Comment by rmarinho** [comment]
build

**Pull request merged**